src/ZoneServer/World/Actors/CombatEntities/Components/RecoveryComponent.cs

```diff
@@ -130,7 +132,21 @@ private void RecoverStamina()
 			if (character.Movement.IsMoving)
 			{
 				var runDrain = (int)character.Properties.GetFloat(PropertyName.Sta_Run, 0);
-				stamina = Math2.Clamp(0, maxStamina, stamina - runDrain);
+
+				// This serves as the update for Dashrun, since it doesn't call its update normally
```

Comment:
This comment appears to be a bit cryptic. It sounds like you're updating `DashRun` in some way, but what you're actually doing seems to be starting a buff if the character is dashing. Do I have that right? Can you tell me more about the requirements? It seems a little disconnected to put this into the stamina recovery. Wouldn't it make more sense in `DashRun.WhileActive` for example?

Reply:
I was originally going to put it in Dashrun.whileActive, as I agree that makes much more sense, but the code there is actually never called because DashRun's updatetime is 0, that's why this component does what is essentially the Dashrun update logic.

Reply:
Hm, so that's basically a bug, because we assumed that that code would run. Do you think giving DashRun an update time could've been a cleaner implementation and bug fix in one? Seems like it would be far less hacky.

Reply:
That's certainly also an option, though it's worth noting that the Dashrun update code currently doesn't execute at all, so we'll likely need to refactor this code to make sure it does what we expect.